### PR TITLE
Adding a getUser api

### PR DESCRIPTION
**Pull request by jpburbank**

### CODE REVIEW

File: src_server/model/ProtocolRouter.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import {ProtocolBiteStep} from './ProtocolBiteStep';
 import {ProtocolRoutingStepState} from './ProtocolRoutingStepState';
+import {ProtocolJumpCodes} from './ProtocolJumpCodes';
```
Comment (jpburbank) [inline src_server/model/ProtocolRouter.ts]: Unrelated to the purpose of the branch, but required to solve a compilation error when running build:compile. Its likely that you too have made this fix or have made other changes that make this moot. If the later, the code difference should get resolved when your changes override it, if you check merge after this merge.

File: src_server/routes/users.ts [hunk 2]
```diff
@@ -19,81 +23,94 @@ const router = express.Router();
...
+  async (req, res, next) => {
+    console.log('*** get all users route, requesting user: ', JSON.stringify(req.user));
 
-  const users = await connection.manager.find(User);
-  res.json({users})
+    const users = await connection.manager.find(User);
+    res.json({users})
 });
 
-router.get('/:id/screeningState', async (req, res, next) => {
```
Comment (jpburbank) [inline src_server/routes/users.ts]: This stub api /:id/screeningState looked like something made moot by sreeningQuestionnaire.ts /:userId/state, so I cleared this out. If wrong I'll back that out.

File: src_server/app.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ const start = async () => {
   // user CRUD
   app.use('/api/users',
     requireJWT,
-    roleAuthorization([RolesEnum.Admin]),
+    roleAuthorization([RolesEnum.Admin, RolesEnum.Caregiver]),
```
Comment (jpburbank) [inline src_server/app.ts]: While I added the role Caregiver I did protect all of the api in /api/user with roleAuthorization([RolesEnum.Admin]) with the exception of the new getUser api. It is protected with your utility function currentUserAuthorizedToTargetUser as the caregiver user needs to request their own user date.

File: src_server/routes/users.ts (hunk 2 above)
Comment (abcd-ca) [reply src_server/routes/users.ts]: We added `state` to the user object in the auth response. You'll still need this for the admin view where you're listing the status of users

File: src_server/app.ts (hunk 3 above)
Comment (abcd-ca) [reply src_server/app.ts]: OK

**Pull request merged**